runtime/src/main/java/com/flipkart/flux/guice/interceptor/TransactionInterceptor.java

```diff
@@ -54,30 +73,22 @@ public Object invoke(MethodInvocation invocation) throws Throwable {
 
             if (transaction != null) {
                 transaction.commit();
```

Comment:
It won't as at the time of method2's execution, variable transaction would be null as it is local to invoke method. Only the topmost method's transaction variable would be not null.